R/select.R

```diff
@@ -39,3 +39,48 @@ select.hyperSpec <- function(.data, ...) {
     .data
   }
 }
+
+# Begin unit testing (UT)
+.test(select.hyperSpec) <- function(){
+  context("select")
+
+  # UT1
+  test_that("selecting data columns correctly", {
+    df <- data.frame(a = NA, b = NA)
+    expect_identical(
+      select(df, a),
+      data.frame(a = NA)
+    )
+  })
+
+  # UT2
+  test_that("using select to rename data columns", {
+    df <- data.frame(a = NA, b = NA)
+    expect_identical(
+      select(df, a_newcolname = a),
+      data.frame(a_newcolname = NA)
+    )
+  })
+
+  # UT3
+  test_that("dropping spectra column", {
+    tmp <- chondro@data
+    tmp$spc <- NULL
+    expect_equivalent(
```

Comment:
(This is the one really **important** comment in this code review:)

You use `expect_equivalent()` rather than `expect_equal()` - presumably because the test would not pass with `expect_equal()`.   
The issue I see here at the moment is that a single `expect_equivalent()` is too lax to give a good test for `select()`. As an example, consider that the subtest for `select(chondro, filename, clusters)` contains a typo (the 2nd column of the resulting data.frame should be `$clusters`, not `$y`) and in consequence _should_ fail right now. But it doesn't. 

What to do? Here are some alternatives (in the order of preference):

1. add the `"labels"` attribute to the manually constructed result data.frame and use `expect_equal()`.

2. In addition to the `expect_equivalent()` which checks the contents of the resulting data.frame, add two more expectations for crucial behaviour of the result:  an expectation for the column names, and an expectation for the `"labels"` attribute because `select.hyperSpec()` explicitly sets this attribute. 

3. Modify the result of `select.hyperSpec()` so that `expect_equal()` can be used. IMHO, this already needs very careful consideration, i.e. a reason why not to use one of the alternatives above in this particular case.

4. Use `expect_equivalent()` for a weak check (as it is now)

Reply:
I have considered your suggestion .3 and it has led me down the following line of thinking:

-> expect_equal checks for equality by comparing values and attributes. 
-> select.hyperSpec ensures that the result has the correct attributes by assigning it the labels of the parameter passed in. 

Which led me to ask the question:

Why is it necessary to assign ALL the labels of the parameter to the result when users are selecting only a subset of the data.frame ```select(chondro, x, y)```? To be clearer, the result generated from ```select(chondro, x, y)``` would still have the labels "filename", "clusters", "spc", and "wavelength", but shouldn't it only have the labels "x" and "y"?

Reply:
You are right that it should not have `labels$filename`  nor `labels$clusters` - good catch! 

Whether labels `labels$.wavelength` and `labels$spc` should be retained (as the labels belonging to the 0 x 0 matrix `@data$spc` is debatable. I suggest to return labels that are consistent with what `chondro [, c("x", "y")]` does: that retains the `.wavelength` label, but not the `spc` label. I'll push a fix.